python/lsst/ts/hvac/hvac_csc.py

```diff
@@ -201,36 +235,20 @@ def _compute_statistics_and_send_telemetry(self):
         for topic in self.hvac_state:
             data = {}
             for item in self.hvac_state[topic]:
-                if not self.hvac_state[topic][item].recent_values:
-                    self.hvac_state[topic][item].recent_median = self.hvac_state[topic][
-                        item
-                    ].initial_value
-                else:
-                    if self.hvac_state[topic][item].data_type == "float":
-                        self.hvac_state[topic][item].recent_median = np.median(
-                            self.hvac_state[topic][item].recent_values
-                        )
-
-                if self.hvac_state[topic][item].initial_value is None:
-                    self.log.error(f"{topic}/{item}: {self.hvac_state[topic][item]}")
-                    continue
-
-                self.hvac_state[topic][item].recent_values = []
                 data[TelemetryItem(item).name] = self.hvac_state[topic][
                     item
-                ].recent_median
+                ].compute_recent_median()
 
             if data:
                 telemetry_method = getattr(self, "tel_" + HvacTopic(topic).name)
                 telemetry_method.set_put(**data)
-                self.log.info(f"Telemetry sent to {telemetry_method}")
 
     def _handle_mqtt_messages(self):
         msgs = self.mqtt_client.msgs
         while not len(msgs) == 0:
             msg = msgs.popleft()
             topic_and_item = msg.topic
-            payload = msg.payload.decode()
+            payload = msg.payload
```

Comment:
At line 270 below I think you have a bug, or at least a misfeature, because if initial_value is None then you set that field but do not append the value to recent_values. Surely you should always append the value to recent_values, regardless of the value of `initial_value`?

I would also argue that you don't need to set `initial_value` at all, at least if type = float, because recent_values will have > 0 values so median will compute a new value and you won't need initial_value.
I still don't understand how non-float values work, but you can always use the first value in the list as the "consolidated" value to use, so I still think you can get rid of initial_value in that case.

The minimal change is:

```
if item_state.initial_value is None:
    item_state.initial_value = value
item_state.recent_values.append(value)
```
and if you do decide you can dump initial_value then it reduces to just the last line.

Reply:
I added the `msgs = self.mqtt_client.msgs` line to reduce what I consider boilerplate code. The idea indeed is to modify the original. However, I replaced the references to `msgs` with `self.mqtt_client.msgs` as you suggested.

The initial_value remark has become obsolete since I removed that as you suggested.

Reply:
Regarding the safety check around (the old) line 261, I added that to check and verify that the Excel sheet that I received, which I converted into a CSV file which acts as an input for all the code, is complete. I removed the check and will add it again when I need it. I just need to remember to remove it when I push code to GitHub.